test/TwigEnvironmentFactoryTest.php

```diff
@@ -81,52 +77,83 @@ public function testUsesDebugConfiguration()
     {
         $config = ['templates' => ['cache_dir' => __DIR__]];
-        $this->container->has('config')->willReturn(true);
-        $this->container->get('config')->willReturn($config);
-        $this->container->has(TwigExtension::class)->willReturn(false);
-        $this->container->has(\Zend\Expressive\Twig\TwigExtension::class)->willReturn(false);
-        $this->container->has(ServerUrlHelper::class)->willReturn(false);
-        $this->container->has(\Zend\Expressive\Helper\ServerUrlHelper::class)->willReturn(false);
-        $this->container->has(UrlHelper::class)->willReturn(false);
-        $this->container->has(\Zend\Expressive\Helper\UrlHelper::class)->willReturn(false);
+        $this->container->expects(self::exactly(2))->method('has')->withConsecutive(
+            ['config'],
+            [TwigExtension::class],
+        )->willReturnOnConsecutiveCalls(true, false);
+        $this->container->expects(self::once())->method('get')->with('config')->willReturn($config);
         $factory     = new TwigEnvironmentFactory();
-        $environment = $factory($this->container->reveal());
+        $environment = $factory($this->container);
 
         $this->assertEquals($config['templates']['cache_dir'], $environment->getCache());
     }
 
-    public function testAddsTwigExtensionIfRouterIsInContainer()
+    /**
+     * @throws LoaderError
+     */
+    public function testAddsTwigExtensionIfRouterIsInContainer(): void
     {
         $twigExtensionFactory = new TwigExtensionFactory();
-        $serverUrlHelper = $this->prophesize(ServerUrlHelper::class)->reveal();
-        $urlHelper       = $this->prophesize(UrlHelper::class)->reveal();
-        $this->container->has('config')->willReturn(false);
-        $this->container->has(ServerUrlHelper::class)->willReturn(true);
-        $this->container->get(ServerUrlHelper::class)->willReturn($serverUrlHelper);
-        $this->container->has(UrlHelper::class)->willReturn(true);
-        $this->container->get(UrlHelper::class)->willReturn($urlHelper);
-        $this->container->has(TwigExtension::class)->willReturn(true);
-        $this->container->get(TwigExtension::class)->willReturn($twigExtensionFactory($this->container->reveal()));
+        $serverUrlHelper      = $this->createMock(ServerUrlHelper::class);
+        $urlHelper            = $this->createMock(UrlHelper::class);
+
+        $this->container->expects(self::exactly(9))->method('has')->withConsecutive(
+            ['config'],
+            [TwigExtension::class],
+            [ServerUrlHelper::class],
+            [UrlHelper::class],
+            [ServerUrlHelper::class],
+            [UrlHelper::class],
+            ['config'],
+            [ServerUrlHelper::class],
+            [UrlHelper::class],
+        )->willReturnOnConsecutiveCalls(
+            false,
+            true,
+            true,
+            true,
+            true,
+            true,
+            false,
+            true,
+            true,
+        );
```

Comment:
Call order does not matter. This test will break if order of calls changes, which makes test brittle as compared to prophecy mock it replaces.

Could you refactor this and other similar changes to use a map or callback instead?

Reply:
Yes that's a much better idea, will do.

Reply:
Refactored all uses of the new mocks to make the tests less brittle as suggested.